pal/src/thread/pal_thread.cpp

```diff
@@ -61,6 +45,33 @@ extern "C" int _lwp_self ();
 
 using namespace CorUnix;
 
+#ifdef __APPLE__
+#define EXPECTED_ALIGNMENT 16 * 1024
+static void GetInternalStackLimit(pthread_t &thread, ULONG_PTR *highLimit, ULONG_PTR *lowLimit)
+{
+    size_t stack = pthread_get_stacksize_np(thread);
+
+    // osx 10.9(+ ?) pthread bug
+    if (pthread_main_np())
+    {
+#ifndef __IOS__
+        stack = max(8 * 1024 * 1024, stack);
```

Comment:
That link is helpful, consider adding to comment.

How about the pthread bug? A link would be helpful.